src/tests/ftest/util/dmg_utils.py

```diff
@@ -792,70 +791,49 @@ def system_query(self, ranks=None, verbose=True):
-            #   Rank UUID                                 Control Address State
-            #   ---- ----                                 --------------- -----
-            #   0    385af2f9-1863-406c-ae94-bffdcd02f379 10.8.1.10:10001 Joined
-            #   1    d7a69a41-59a2-4dec-a620-a52217851285 10.8.1.11:10001 Joined
-            #   Rank UUID   Control Address  Fault Domain  State  Reason
-            #   ---- ----   ---------------  ------------  -----  ------
-            #   0    <uuid> <address>        <domain>      Joined system stop
-            #   1    <uuid> <address>        <domain>      Joined system stop
-            #
-            #       Where the above placeholders have values similar to:
-            #           <uuid>    = 0c21d700-0e2b-46fb-be49-1fca490ce5b0
-            #           <address> = 10.8.1.142:10001
-            #           <domain>  = /wolf-142.wolf.hpdd.intel.com
-            #
-            match = re.findall(
-                r"(\d+)\s+([0-9a-f-]+)\s+([0-9.:]+)\s+([/A-Za-z0-9-_.]+)"
-                r"\s+([A-Za-z]+)(.*)",
-                self.result.stdout)
-            for info in match:
-                data[int(info[0])] = {
-                    "uuid": info[1],
-                    "address": info[2],
-                    "domain": info[3],
-                    "state": info[4],
-                    "reason": info[5].strip(),
+        # Sample output:
+        # {
+        # "response": {
+        #     "members": [
+        #     {
+        #         "addr": "10.8.1.11:10001",
+        #         "state": "joined",
+        #         "fault_domain": "/wolf-11.wolf.hpdd.intel.com",
+        #         "rank": 0,
+        #         "uuid": "e7f2cb06-a111-4d55-a6a5-b494b70d62ab",
+        #         "fabric_uri": "ofi+sockets://192.168.100.11:31416",
+        #         "fabric_contexts": 17,
+        #         "info": ""
+        #     },
+        #     {
+        #         "addr": "10.8.1.74:10001",
+        #         "state": "evicted",
+        #         "fault_domain": "/wolf-74.wolf.hpdd.intel.com",
+        #         "rank": 1,
+        #         "uuid": "db36ab28-fdb0-4822-97e6-89547393ed03",
+        #         "fabric_uri": "ofi+sockets://192.168.100.74:31416",
+        #         "fabric_contexts": 17,
+        #         "info": ""
+        #     }
+        #     ]
+        # },
+        # "error": null,
+        # "status": 0
+        # }
+        output = self._get_json_result(
+            ("system", "query"), ranks=ranks, verbose=verbose)
+        if "response" in output and "members" in output["response"]:
+            for member in output["response"]["members"]:
+                data[member["rank"]] = {
+                    "address": member["addr"],
+                    "uuid": member["uuid"],
+                    "domain": member["fault_domain"],
+                    "state": member["state"],
+                    "reason": member["info"],
```

Comment:
```suggestion
                    "info": member["info"],
```
Is this typo?

Reply:
No. The current dictionary returned by this method uses the "reason" key.  It would be a more extensive change to align the keys and if we're going to put the effort in we really should update the consumers of this method to accept the unaltered json output.

Reply:
I see. Thanks.

Reply:
Due to changes in https://github.com/daos-stack/daos/pull/4562, I've updated the code consuming this method to accept the unaltered json output.